### PR TITLE
Support pure Unicode emojis as a lookup for reactions.

### PR DESCRIPTION
**Pull request by EpsilonPrime**
# Description
Fixes issue where requesting a unicode-based Emoji using an Emoji struct would not work:

`
    reactors, err := discord.Channel(channel.ID).Message(msg.ID).React  ion(reaction.Emoji).Get(&disgord.GetReactionURLParams{Limit: 100})
`

Two sample emojis (the first is a managed one, the second is unicode-based):
`emoji{name:sigma, id:532411072661946391}`
`emoji{name:👌, id:0}`

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [ ] I ran `go generate`
- [x] I have performed a self-review of my own code (remember to run `go fmt ./...`)
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Added benchmarks if this is a performant required component (potential bottlenecks)

### CODE REVIEW

File: reaction.go [hunk 1]
```diff
@@ -242,7 +242,11 @@ func (r reactionQueryBuilder) Get(params URLQueryStringer, flags ...Flag) (ret [
 
 	emojiCode := ""
 	if e, ok := r.emoji.(*Emoji); ok {
-		emojiCode = e.Name + ":" + e.ID.String()
+		if e.ID != 0 {
```
Comment (andersfylling) [inline reaction.go:245]: Could you use e.ID.IsZero() instead?
just for consistency and feature behaviour patches

**Comment by andersfylling** [comment]
IsZero is the same as `e.ID == 0`. There are also several other functions that utilize the same logic as this. Could you patch up those as well?

**Comment by andersfylling** [comment]
I would suggest making a function that can be re-used across these REST methods

**Comment by EpsilonPrime** [comment]
Created two new functions to reduce the duplication in the file.

**Pull request merged**